src/X.PagedList/OrderedPagedList.cs

```diff
@@ -39,29 +39,25 @@ public class PagedList<T, TKey> : BasePagedList<T>
     public PagedList(IQueryable<T>? superset, Expression<Func<T, TKey>> keySelector, int pageNumber, int pageSize)
         : base(pageNumber, pageSize, superset?.Count() ?? 0)
     {
-        // add items to internal list
-        if (superset != null && TotalItemCount > 0)
-        {
-            InitSubset(superset, keySelector.Compile(), pageNumber, pageSize);
-        }
+        InitSubset(superset, keySelector.Compile(), pageNumber, pageSize);
```

Comment:
Note that there isn't really much point to creating a PagedList without a superset.

Personally, I would make the superset argument here non-nullable, throw an exception if it is, and drop the nullable handling after that point.

If somebody really wants a PagedList that is empty, he could still supply an empty list to the constructor. This would look like this:

```
public PagedList(IQueryable<T> superset, Expression<Func<T, TKey>> keySelector, int pageNumber, int pageSize)
        : base(pageNumber, pageSize, superset.Count())
    {
        // add items to internal list
        if (TotalItemCount > 0)
        {
            InitSubset(superset, keySelector.Compile(), pageNumber, pageSize);
        }
    }
```

I would do the same to equivalent cases, i.e. never allow null for the superset anywhere.
This also means that you do not have to change InitSubset at all.

Reply:
Agree